boa/interpret.py

```diff
@@ -160,6 +164,10 @@ def get_compiler_data():
         with anchor_settings(ret.settings):
             # force compilation to happen so DiskCache will cache the compiled artifact:
             _ = ret.bytecode, ret.bytecode_runtime
+
+        if isinstance(ret.input_bundle, ZipInputBundle):
+            # workaround for `cannot pickle '_thread.RLock' object`
+            ret.input_bundle.archive._lock = None
```

Comment:
kind of a weird one. why is there an active lock at this stage?

Reply:
It doesn't need to be active to be unpickable. ZipFile seems to create a lock on construction: https://github.com/python/cpython/blob/850189a64e7f0b920fe48cb12a5da3e648435680/Lib/zipfile.py#L1266

Reply:
> however, on the boa side we cannot make `CompileData` pickable.

what do you mean here?